docs/src/details/xml/editor/user_interface.rst

```diff
@@ -0,0 +1,172 @@
+.. include:: substitutions.txt
+.. _user interface:
+
+==============
+User Interface
+==============
+
+
+
+First Run
+*********
+
+The first time you launch |lvglpro|, it will pop up a window with the message
+
+    "Pulling Container image for LVGL, this may take a few minutes...".
+
+This message means the Editor is downloading a pre-configured a container image that
+includes all the necessary tools and dependencies for building LVGL projects.
+|lvglpro| handles (and updates) the container automatically in the background.  The
+process can take a few minutes the first time, depending on your internet speed, but
+it only happens once.  After that, the container is reused for future sessions.
+
+When a new version of the container is published (e.g. with updated toolchains, build
+logic, etc.), the Editor automatically checks for and pulls the latest version.  You
+won’t need to do anything manually, it stays up-to-date in the background.
+
+
+
+Initial Display
+***************
+
+.. figure:: /_static/images/editor_initial_ui.png
+    :align: center
+    :alt: |lvglpro| Opening User Interface
+
+    ..
+
+    |lvglpro| Opening User Interface
+
+
+Component Libraries vs Regular Projects
+---------------------------------------
+
+A Component Library is a set of 1 or more XML files that can be used in any
+|lvglpro| Project.
+
+Regular |lvglpro| Projects use Component Libraries, but Component Libraries do not
+use regular Projects.
+
+
+The Difference Between the Two
+------------------------------
+
+The difference between the directory structures two types of |lvglpro| projects
+(Projects vs Component Libraries) is that a Project has a ``project.xml`` file in it
+(which ties together all of its parts), and a Component Library does not.
+
+TODO:  elaborate on what ``project.xml`` describes Gabor
+
+
+Nesting Projects
+----------------
+
+TODO Gabor
```

Comment:
We will simplify it by v1.0 and there won't be component library button. 

As it's an advanced use case I'll explain how to create nested project somewhere else.